File: statsig-lib/src/statsig_core_api_options.rs
use serde::Deserialize;

pub enum AnyEvaluationOptions {
    FeatureGateEvaluationOptions(FeatureGateEvaluationOptions),
    DynamicConfigEvaluationOptions(DynamicConfigEvaluationOptions),
    ExperimentEvaluationOptions(ExperimentEvaluationOptions),
    LayerEvaluationOptions(LayerEvaluationOptions),
}

#[derive(Debug, Clone, Default, Deserialize)]
pub struct FeatureGateEvaluationOptions {
    #[serde(default)]
    pub disable_exposure_logging: bool,
}

#[derive(Debug, Clone, Default, Deserialize)]
pub struct DynamicConfigEvaluationOptions {
    #[serde(default)]
    pub disable_exposure_logging: bool,
}

#[derive(Debug, Clone, Default, Deserialize)]
pub struct ExperimentEvaluationOptions {
    #[serde(default)]
    pub disable_exposure_logging: bool,
}

#[derive(Debug, Clone, Default, Deserialize)]
pub struct LayerEvaluationOptions {
    #[serde(default)]
    pub disable_exposure_logging: bool,
}

File: statsig-go/statsig_core_api_options.go
package statsig_go_core

type FeatureGateEvaluationOptions struct {
	DisableExposureLogging bool `json:"disable_exposure_logging"`
}

type DynamicConfigEvaluationOptions struct {
	DisableExposureLogging bool `json:"disable_exposure_logging"`
}

type ExperimentEvaluationOptions struct {
	DisableExposureLogging bool                 `json:"disable_exposure_logging"`
	UserPersistedValues    *UserPersistedValues `json:"user_persisted_values,omitempty"`
}

type LayerEvaluationOptions struct {
	DisableExposureLogging bool                 `json:"disable_exposure_logging"`
	UserPersistedValues    *UserPersistedValues `json:"user_persisted_values,omitempty"`
}

type ParameterStoreEvaluationOptions struct {
	DisableExposureLogging bool `json:"disable_exposure_logging"`
}
